File: data/src/main/java/home/oleg/placesnearme/mapper/PhotoMapper.java
package home.oleg.placesnearme.mapper;

import java.util.ArrayList;
import java.util.List;

import home.oleg.placenearme.models.Photo;

/**
 * Created by Oleg Sheliakin on 25.09.2018.
 * Contact me by email - olegsheliakin@gmail.com
 */
public final class PhotoMapper {

    private PhotoMapper() {
    }

    public static Photo map(home.oleg.placesnearme.core_network.models.Photo photo) {
        Photo result = new Photo();

        result.setCreatedAt(photo.getCreatedAt());
        result.setHeight(photo.getHeight());
        result.setId(photo.getId());
        result.setPrefix(photo.getPrefix());
        result.setSuffix(photo.getSuffix());
        result.setWidth(photo.getWidth());
        result.setVisibility(photo.getVisibility());
        return result;
    }

    public static List<Photo> map(List<home.oleg.placesnearme.core_network.models.Photo> photos) {
        List<Photo> result = new ArrayList<>();

        for (home.oleg.placesnearme.core_network.models.Photo photo : photos) {
            result.add(map(photo));
        }

        return result;
    }
}

File: core-data/src/main/java/home/oleg/placesnearme/coredata/mapper/PhotoMapper.kt
package home.oleg.placesnearme.coredata.mapper

import home.oleg.placesnearme.coredata.model.PhotoEntity
import home.oleg.placesnearme.coredomain.models.Photo
import java.util.*

/**
 * Created by Oleg Sheliakin on 25.09.2018.
 * Contact me by email - olegsheliakin@gmail.com
 */
object PhotoMapper {

    fun map(photo: home.oleg.placesnearme.corenetwork.models.Photo) = Photo(
            height = photo.height,
            suffix = photo.suffix,
            prefix = photo.prefix,
            width = photo.width
    )

    fun map(photos: List<home.oleg.placesnearme.corenetwork.models.Photo>): List<Photo> {
        val result = ArrayList<Photo>()

        for (photo in photos) {
            result.add(map(photo))
        }

        return result
    }

    fun mapFromDb(photos: List<PhotoEntity>): List<Photo> {
        val result = ArrayList<Photo>()

        for (photo in photos) {
            result.add(photo.photo)
        }

        return result
    }

    fun mapToDb(photos: List<Photo>): List<PhotoEntity> {
        val result = ArrayList<PhotoEntity>()

        for (photo in photos) {
            val photoDbEntity = PhotoEntity(
                    photo = photo
            )
            result.add(photoDbEntity)
        }

        return result
    }
}
